File: admin/src/utils/apiError.js
export function parseApiError(err, fallback = 'Something went wrong') {
  const data = err?.response?.data;
  if (data?.errors?.length) {
    return data.errors.map((e) => e.message).join('. ');
  }
  if (data?.message && data.message !== 'Validation error') {
    return data.message;
  }
  if (data?.message) {
    return data.message;
  }
  return fallback;
}

File: mobile/lib/core/utils/api_error.dart
import 'package:dio/dio.dart';

String parseApiError(dynamic error) {
  if (error is DioException) {
    final data = error.response?.data;
    if (data is Map) {
      final errors = data['errors'];
      if (errors is List && errors.isNotEmpty) {
        return errors
            .map((e) => e is Map ? (e['message']?.toString() ?? '') : e.toString())
            .where((m) => m.isNotEmpty)
            .join('. ');
      }
      if (data['message'] != null) {
        return data['message'].toString();
      }
    }
    switch (error.type) {
      case DioExceptionType.connectionTimeout:
      case DioExceptionType.receiveTimeout:
        return 'Connection timed out. Please try again.';
      case DioExceptionType.connectionError:
        return 'No internet connection. Check your network.';
      default:
        return error.message ?? 'Something went wrong';
    }
  }
  return error.toString();
}
